ingestion_tools/scripts/data_validation/run_tests.py

```diff
@@ -10,79 +9,95 @@
+    local_dir: str,
+    output_dir: str,
+    input_bucket: str,
     output_bucket: str,
     datasets: str | None,
-    save_history: bool,
+    multiprocessing: bool,
+    history: bool,
     extra_args: str | None,
 ):
     fs: S3Filesystem = FileSystemApi.get_fs_api(mode="s3", force_overwrite=False)
-    now = datetime.datetime.now().isoformat(sep="_", timespec="seconds").replace(":", "-")
 
     if datasets is None:
-        datasets = fs.glob(f"s3://{bucket}/*")
+        datasets = fs.glob(f"s3://{input_bucket}/*")
         datasets = [os.path.basename(dataset) for dataset in datasets if os.path.basename(dataset).isdigit()]
     else:
         datasets = datasets.split(",")
 
+    datasets = sorted(datasets, key=lambda x: int(x))
+
+    # If we are not saving history, we can run all datasets in one go (individual dataset runs don't need to be tracked)
+    if not history:
+        datasets = [",".join(datasets)]
+
     extra_args = extra_args if extra_args else ""
 
     for dataset in datasets:
+        now = datetime.datetime.now().isoformat(sep="_", timespec="seconds").replace(":", "-")
+
         # Accumulate test results here
-        localdir_raw = f"{local_dst}/{dataset}_raw/{now}"
+        localdir_raw = f"{local_dir}/{dataset}_raw/{now}"
         # Generate report here
-        localdir_rep = f"{local_dst}/{dataset}/{now}"
-        # Archive location
-        local_tar = f"{local_dst}/{dataset}/{dataset}_{now}.tar.gz"
-
-        os.makedirs(localdir_raw, exist_ok=True)
-        os.makedirs(localdir_rep, exist_ok=True)
+        localdir_rep = f"{local_dir}/{dataset}/{now}"
 
         # Run tests and generate results
-        os.system(f"pytest -n auto --dist worksteal --dataset {dataset} --alluredir {localdir_raw} {extra_args}")
+        os.system(
+            f"pytest {'--dist worksteal -n auto' if multiprocessing else '--dist no'} --datasets {dataset} --alluredir {localdir_raw} {extra_args}",
+        )
 
         # Get the history from S3 (Must do this before generating the report)
-        remote_dataset_dir = f"{output_bucket}/data_validation/{dataset}"
-        remote_dataset_tar = f"{remote_dataset_dir}/{dataset}.tar.gz"
-        fs.makedirs(remote_dataset_dir)
-
-        if fs.exists(remote_dataset_tar):
-            get_history(remote_dataset_tar, localdir_raw, fs)
+        remote_dataset_dir = f"{output_bucket}/{output_dir}/{dataset}"
+        if fs.exists(remote_dataset_dir):
+            get_history(remote_dataset_dir, localdir_raw, fs)
+            fs.s3fs.rm(remote_dataset_dir, recursive=True)
 
         # Generate the report
         os.system(f"allure generate --output {localdir_rep} {localdir_raw}")
 
         # Upload the new report
-        if save_history:
-            os.system(f"tar -czf {local_tar} -C {localdir_rep} .")
-            fs.s3fs.put(local_tar, remote_dataset_tar)
-            os.system(f"rm -rf {local_tar}")
+        fs.s3fs.put(localdir_rep, remote_dataset_dir, recursive=True)
```

Comment:
Why not use `fs.push`?

Reply:
fs.push requires it to be in the temp dir, but the user may want to store it somewhere other than the tmp dir?